### ISSUE TITLE
Training pi0 from scratch

### ISSUE THREAD

**Issue by RhythmSyed** [Physical-Intelligence/openpi#436]
Im interested in training pi0 from scratch, where can I access the pretraining cross-embodiment data and how should I go about training the model from the skeleton VLM and action expert?

**Comment by kpertsch** [comment]
Hi!

Thanks for your interest in pi0! The openpi codebase is currently mainly designed for model fine-tuning. For pre-training, you may encounter some scalability issues (eg LeRobot datasets, which we use for data loading, are not currently designed for representing very large datasets). That being said, it should be possible to use openpi for model pre-training. You can load pre-trained paligemma weights into the model during initialization (instead of the pi0 weights), and then use the same finetuning routine we use for pi0-finetuning, just on a broader robot pre-training dataset.

Regarding data: not all of the data in the pi0 pre-training mixture is currently publicly available. However, if you are interested in pre-training, I would encourage you to look into [Open X-Embodiment ](https://robotics-transformer-x.github.io/) and the [DROID dataset](https://droid-dataset.github.io/). Both are part of our pre-training mixture, and there are codebases that show how to load from them efficiently for model pre-training (eg check out the [OpenVLA](https://openvla.github.io/) codebase).

Hope this helps! I will close this issue for now, but feel free to reopen!

**Comment by zstarN70** [comment]
Thanks for your excellent work! @kpertsch 

I’d like to reproduce the training of the Pi-0 model (or similar architecture). Could you kindly provide the detailed training recipe?

For example:
- Number of GPUs
- Total batch size
- Training duration (steps or days)
- Optimizer settings
- Gradient accumulation steps (if any)
- Dataset augmentation
- Image resolution
- Any special tricks (like EMA, classifier-free guidance etc.)

Thanks a lot!

**Comment by LogSSim** [comment]
> Hi!
> 
> Thanks for your interest in pi0! The openpi codebase is currently mainly designed for model fine-tuning. For pre-training, you may encounter some scalability issues (eg LeRobot datasets, which we use for data loading, are not currently designed for representing very large datasets). That being said, it should be possible to use openpi for model pre-training. You can load pre-trained paligemma weights into the model during initialization (instead of the pi0 weights), and then use the same finetuning routine we use for pi0-finetuning, just on a broader robot pre-training dataset.
> 
> Regarding data: not all of the data in the pi0 pre-training mixture is currently publicly available. However, if you are interested in pre-training, I would encourage you to look into [Open X-Embodiment ](https://robotics-transformer-x.github.io/) and the [DROID dataset](https://droid-dataset.github.io/). Both are part of our pre-training mixture, and there are codebases that show how to load from them efficiently for model pre-training (eg check out the [OpenVLA](https://openvla.github.io/) codebase).
> 
> Hope this helps! I will close this issue for now, but feel free to reopen!

I would like to know what the version parameters of Paligemma are used for in pi0?
